common/geth/multihoming_client.go

```diff
@@ -0,0 +1,551 @@
+package geth
```

Comment:
There should be `multihoming_client_test.go` that creates several mocked EthClients and tests if they're cycled properly under various conditions (a condition where a call returns 200 status with json-rpc error, one which returns 5xx error, 429 error, etc)

Reply:
good call, I added some new tests for the multihoming clients, but they are based on EVM code, not json error.